src/accelerate/utils/modeling.py

```diff
@@ -1267,8 +1308,33 @@ def infer_auto_device_map(
             current_memory_used += module_size
             device_map[name] = devices[current_device]
 
+            if not offload_buffers:
+                if device not in device_buffer_sizes.keys():
+                    device_buffer_sizes[device] = 0
+
+                current_buffer_size = compute_module_total_buffer_size(
+                    module, dtype=dtype, special_dtypes=special_dtypes
+                )
+                device_buffer_sizes[device] = device_buffer_sizes[device] + current_buffer_size
+
     if clean_result:
         device_map = clean_device_map(device_map)
+
+    if not offload_buffers:
+        non_gpu_buffer_size = device_buffer_sizes.get("cpu", 0) + device_buffer_sizes.get("disk", 0)
+
+        for gpu_device, gpu_max_memory in max_memory.items():
+            if gpu_device == "cpu" or gpu_device == "disk":
+                continue
+
+            gpu_memory_used = device_memory_used.get(gpu_device, 0)
+            if gpu_max_memory < non_gpu_buffer_size + gpu_memory_used:
+                raise ValueError(
+                    f"Device {gpu_device} has {gpu_max_memory} bytes of memory, and has planned to use {gpu_memory_used} bytes, "
+                    f"so the remaining memory is not enough to store additional buffers ({non_gpu_buffer_size} bytes)! "
+                    f"Please consider using offload_buffers=True."
+                )
```

Comment:
This will trigger an error even when we have, for example, the first gpu being free but the second gpu is full. The buffers can potentially go on the first gpu. When we don't offload the buffers, they are on the execution device (can be any device). Maybe we can just put a warning to let the user know that the buffer that won't be offloaded and it won't probably fit on one of the gpus.

Reply:
Yes, I did not consider the multi-gpu scenario because I have only one card. I'll change this to a warning instead of an `Error`. I will also try to determine the execution device more precisely to see if we can make this warning be more useful.

Reply:
btw, could you please give some hint on the code which implemented the "the first gpu being free but the second gpu is full. The buffers can potentially go on the first gpu"? The only thing I found is the `AlignDevicesHook` class in `src/accelerate/hooks.py`, which seems moving all buffers to all execution devices.

Reply:
Basically, what I mean is if the user let a `device_map` that only most of the modules on the second gpus and leave of the space available on the first gpu, the buffer will most likely fit in the first gpu (if we suppose the first gpu is the execution device). For example, this is the case with `device_map="balanced_low_0"` strategy.